pkg/controller/storagecluster/reconcile.go

```diff
@@ -135,17 +135,25 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		if errors.IsNotFound(err) {
 			reqLogger.Info("Creating StorageClusterInitialization resource")
 
+			// if the StorageClusterInitialization object doesn't exist
+			// ensure we re-reconcile on all initialization resources
+			instance.Status.StorageClassesCreated = false
+			instance.Status.CephObjectStoresCreated = false
+			instance.Status.CephBlockPoolsCreated = false
+			instance.Status.CephObjectStoreUsersCreated = false
+			instance.Status.CephFilesystemsCreated = false
+			instance.Status.FailureDomain = determineFailureDomain(instance.Status.NodeTopologies)
```

Comment:
I think this should be set before we get here, so that it is properly reconciled every time and not just when the SCI isn't present. Specifically, around line 130.

Reply:
I can do that, it appeared that FailureDomain was only relevant during initialization though.